dessia_common/core.py

```diff
@@ -638,10 +638,28 @@ def to_xlsx_stream(self, stream):
         writer.save_to_stream(stream)
 
     def _export_formats(self):
-        formats = [{"extension": "json", "method_name": "save_to_stream", "text": True, "args": {}},
-                   {"extension": "xlsx", "method_name": "to_xlsx_stream", "text": False, "args": {}}]
+        formats = [{"selector": "json", "extension": "json",
+                    "method_name": "save_to_stream", "text": True, "args": {}},
+                   {"selector": "xlsx", "extension": "xlsx",
+                    "method_name": "to_xlsx_stream", "text": False, "args": {}}]
         return formats
 
+    def save_export_to_file(self, selector, filepath):
+        for export_format in self._export_formats():
+            if export_format['selector'] == selector:
+                if not filepath.endswith(f".{export_format['extension']}"):
+                    filepath += f".{export_format['extension']}"
+                    print(f'Renaming filepath to {filepath}')
+                if export_format['text']:
+                    mode = 'w'
+                else:
+                    mode = 'wb'
```

Comment:
Here is a suggestion, not sure it is better
you can do it :
`mode = 'wb'[:1 + bool(export_format['text'])]`

Reply:
Same remark as the last comment, here. I find it really complex for simple stuff.

I personnally think that the right solution here would be : 
```
mode = "w"                              # or whatever you want it to be
if not export_format["text"]:
    mode += "b"                         # or mode = f"{mode}b"
```